app/toolbox.cpp

```diff
@@ -31,164 +31,228 @@ GNU General Public License for more details.
+    penButton->setWhatsThis( tr( "Pen Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_PEN ) ) );
+    eraserButton->setWhatsThis( tr( "Eraser Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_ERASER ) ) );
+    polylineButton->setWhatsThis( tr( "Polyline Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_POLYLINE ) ) );
+    bucketButton->setWhatsThis( tr( "Paint Bucket Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_BUCKET ) ) );
+    colouringButton->setWhatsThis( tr( "Brush Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_BRUSH ) ) );
+    eyedropperButton->setWhatsThis( tr( "Eyedropper Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_EYEDROPPER ) ) );
+    clearButton->setWhatsThis( tr( "Clear Tool (%1)" )
+        .arg( GetToolTips( CMD_CLEAR_FRAME ) ) );
+    smudgeButton->setWhatsThis( tr( "Smudge Tool (%1)" )
+        .arg( GetToolTips( CMD_TOOL_SMUDGE ) ) );
+
+    pencilButton->setCheckable( true );
+    penButton->setCheckable( true );
+    polylineButton->setCheckable( true );
+    bucketButton->setCheckable( true );
+    colouringButton->setCheckable( true );
+    smudgeButton->setCheckable( true );
+    eyedropperButton->setCheckable( true );
+    selectButton->setCheckable( true );
+    moveButton->setCheckable( true );
+    handButton->setCheckable( true );
+    eraserButton->setCheckable( true );
+    pencilButton->setChecked( true );
+
+    layout->setMargin( 0 );
+    layout->setSpacing( 0 );
+
+    QWidget* toolGroup = new QWidget();
+
+    setWidget( toolGroup );
+    toolGroup->setLayout( layout );
+
+    connect( pencilButton, &QToolButton::clicked, this, &ToolBoxWidget::pencilOn );
+    connect( eraserButton, &QToolButton::clicked, this, &ToolBoxWidget::eraserOn );
+    connect( selectButton, &QToolButton::clicked, this, &ToolBoxWidget::selectOn );
+    connect( moveButton, &QToolButton::clicked, this, &ToolBoxWidget::moveOn );
+    connect( penButton, &QToolButton::clicked, this, &ToolBoxWidget::penOn );
+    connect( handButton, &QToolButton::clicked, this, &ToolBoxWidget::handOn );
+    connect( polylineButton, &QToolButton::clicked, this, &ToolBoxWidget::polylineOn );
+    connect( bucketButton, &QToolButton::clicked, this, &ToolBoxWidget::bucketOn );
+    connect( eyedropperButton, &QToolButton::clicked, this, &ToolBoxWidget::eyedropperOn );
+    connect( colouringButton, &QToolButton::clicked, this, &ToolBoxWidget::brushOn );
+    connect( smudgeButton, &QToolButton::clicked, this, &ToolBoxWidget::smudgeOn );
+    connect( clearButton, &QToolButton::clicked, this, &ToolBoxWidget::clearButtonClicked );
+
+    QSettings settings(PENCIL2D, PENCIL2D);
+    this->restoreGeometry(settings.value( "ToolBoxGeom" ).toByteArray());
 }
 
 void ToolBoxWidget::updateUI()
 {
 }
 
+void ToolBoxWidget::resizeEvent(QResizeEvent* event)
+{
+    QRect geom = this->geometry();
+    int SIDEBYSIDELAYOUT = 76;
+    int SINGLEROWLAYOUT = 215;
+
+    if (geom.width() > SIDEBYSIDELAYOUT) {
```

Comment:
This works, but I think it would be even better if we didn't have conditions for specific states, but instead added them in a loop with some way of calculating the "optimal" rows and columns for the particular width and height. I don't know how exactly to determine the best rows/columns, but it would allow for the user to use other layouts (ex. a 6x2, 3x4, or 4x3) which is a nice thing to have.